lib/pubsuffix-psl.js

```diff
@@ -61,7 +61,11 @@ function getPublicSuffix(domain, options = {}) {
     }
   }
 
-  if (!ignoreError && SPECIAL_USE_DOMAINS.includes(topLevelDomain)) {
+  if (
+    !ignoreError &&
+    !allowSpecialUseDomain &&
```

Comment:
What does this change mean? Are we going to return `null` for all domains which are not special treatment? Also what is the difference between returning `null` and `test`?

Reply:
I have changed this `If` statement so that we don't throw any error when `allowSpecialUseDomain` is set to `true`.
We are returning null in the following cases:
- when domain is consist of only one specialUseDomain which is not specialTreatmentDomain
- when domain is not present in public suffix list.

There is no difference between returning `null` or `test` because inside the parent function `findCookies` we convert `null` into `[domain]`.

Reply:
As the name suggests, this function returns the public suffix of the domain, since `test` or any other special use domain does not come under the public suffix [list](https://publicsuffix.org/list/public_suffix_list.dat) so we should return `null` in those cases.